src/citrine/resources/file_link.py

```diff
@@ -57,6 +68,77 @@ def build(self, data: dict) -> FileLink:
         """Build an instance of FileLink."""
         return FileLink.build(data)
 
+    def list(self, page: Optional[int] = None,
+             per_page: Optional[int] = None) -> Iterable[FileLink]:
+        """
+        List all visible files in the collection.
+
+        Parameters
+        ---------
+        page: int, optional
+            The "page" of results to list. Default is the first page, which is 1.
+        per_page: int, optional
+            Max number of results to return. Default is 20.
+
+        Returns
+        -------
+        Iterable[FileLink]
+            FileLink objects in this collection.
+
+        """
+        path = self._get_path()
+        params = {}
+        if page is not None:
+            params["page"] = page
+        if per_page is not None:
+            params["per_page"] = per_page
+
+        response = self.session.get_resource(path=path, params=params)
+        collection = response[self._collection_key]
+        for file in collection:
+            yield self.build(self._as_dict_from_resource(file))
+
+    def _as_dict_from_resource(self, file: dict):
+        """
+        Convert a file link resource downloaded from the API into a FileLink dictionary.
+
+        This is necessary because the database resource contains additional information that is
+        not in the FileLink object, such as file size and the id of the user who uploaded the file.
+
+        Paramters
+        ---------
+        file: dict
+            A JSON dictionary corresponding to the file link as it is saved in the database.
+
+        Returns
+        -------
+        dict
+            A dictionary that can be built into a FileLink object.
+
+        """
+        typ = 'file_link'
+        filename = file['filename']
+
+        # The field 'versioned_url' contains some information necessary to construct a file path,
+        # but does not contain project and dataset id. It also contains extraneous information.
+        # We assert that the 'versioned_url' "picks up" where the collection path leaves off
```

Comment:
Mixed single and double quotes is a little odd

Reply:
The single quotes are a key name, while the double quotes indicate that I'm speaking loosely (Imagine I make quotes with my fingers). I'd love a suggestion for how to make this idea of joining the versioned_url and collection path together. It's not a straight join. The versioned url is `https://a-bunch-of-other-stuff-whatever/files/{file_id}/version/{version}` and the collection path is `projects/{project_id}/datasets/{dataset_id}/files`. I'm snipping the irrelevant bits and pasting them together at "files"